Userland/Libraries/LibJS/Runtime/IndexedProperties.cpp

```diff
@@ -35,8 +35,13 @@ void SimpleIndexedPropertyStorage::grow_storage_if_needed()
 {
     if (m_array_size <= m_packed_elements.size())
         return;
-    // Grow storage by 25% at a time.
-    m_packed_elements.resize(m_array_size + (m_array_size / 4));
+
+    if (m_array_size <= m_packed_elements.capacity()) {
```

Comment:
Usually these allow up to ~75-80% capacity utilisation before growing again, maybe experiment with that too?
(not a requirement or anything, just curious)

Reply:
From my testing, this doesn't seem to improve performace. Although I might not quite understand what you mean. Why exactly would we want to always keep ~20% free?

Reply:
I don't really have a concrete idea why, I've just seen it done a lot 😅

Reply:
We are still a lot slower than v8 and sm but I think we can leave this until we have more suitable benchmarks (I bet they detect this kind of push in a loop anyway)